temporal-sdk/src/main/java/io/temporal/worker/Worker.java

```diff
@@ -323,25 +327,38 @@ void start() {
       return;
     }
     workflowWorker.start();
-    activityWorker.start();
+    if (!options.isActivityPollerDisabled()) {
```

Comment:
I would change all this check to `activityWorker != null` which would make it more obvious.

Reply:
I wanted to use this option explicitly in order to be able to catch potential bugs more easily in case if activityWorker is null when it shouldn't be.